rascal-lsp/src/main/rascal/util/LanguageServer.rsc

```diff
@@ -125,204 +109,132 @@ of the user.
+   * a ((definition)) service is a fast and location specific version of the `definitions` relation in a ((Summary)).
+   * a ((reference)) service is a fast and location specific version of the `references` relation in a ((Summary)).
+   * an ((implementation)) service is a fast and location specific version of the `implementations` relation in a ((Summary)).
+* The ((documentSymbol)) service maps a source file to a pretty hierarchy for visualization in the "outline" view and "symbol search" features.
+* The ((lenses)) service discovers places to add "lenses" (little views embedded in the editor on a separate line) and connects commands to execute to each lense
+* The ((inlayHint)) service discovers plances to add "inlays" (little views embedded in the editor on the same line). Unlike ((lenses)) inlays do not offer command execution.
+* The ((execution)) service executes the commands registered by ((lenses)) and ((inlayHinter))s.
+* The ((actions)) service discovers palces to add "code actions" (little hints in the margin next to where the action is relevant) and connects ((CodeAction))s to execute when the users selects the action from a menu.
 
 Many language contributions received a ((Focus)) parameter. This helps to create functionality that
 is syntax-directed: relevant to the current syntactical constructs under the cursor.
 }
 data LanguageService
-    = parser(Parser parser)
-    | analyzer(Summarizer summarizer
+    = parsing(Tree (str _input, loc _origin) parser)
+    | analysis(Summary (loc _origin, Tree _input) summarizer
         , bool providesDocumentation = true
         , bool providesDefinitions = true
         , bool providesReferences = true
         , bool providesImplementations = true)
-    | builder(Summarizer summarizer
+    | build(Summary (loc _origin, Tree _input) summarizer
         , bool providesDocumentation = true
         , bool providesDefinitions = true
         , bool providesReferences = true
         , bool providesImplementations = true)
-    | outliner(Outliner outliner)
-    | lenses(OrderedLensDetector detector)
-    | inlayHinter(InlayHinter hinter)
-    | executor(CommandExecutor executor)
-    | documenter(FocusDocumenter documenter)
-    | definer(FocusDefiner definer)
-    | referrer(FocusReferrer reference)
-    | implementer(FocusImplementer implementer)
-    | actions(CodeActionContributor actions)
+    | documentSymbol(list[DocumentSymbol] (Tree _input) labeler)
+    | codeLense(lrel[loc src, Command lens] (Tree _input) detector)
+    | inlayHint(list[InlayHint] (Tree _input) hinter)
+    | execution(value (Command _command) executor)
+    | hover(set[str] (Focus _focus) documentor)
+    | definition(set[loc] (Focus _focus) linker)
+    | reference(set[loc] (Focus _focus) linker)
+    | implementation(set[loc] (Focus _focus) linker)
+    | codeAction(list[CodeAction] (Focus _focus) actions)
```

Comment:
Nit: Also rename arguments to be consistent with the contructor? For instance, `linker` as a term doesn't seem to occur anywhere else in this module (not in the docs), so it's not so clear which abstraction/model it belongs to. I'd have a mild preference for `definitionService`, `definitionCallback`, `fDefinition`, `definitionFn`, or something along those lines.

Reply:
I've been having trouble with these names since forever. They are inconsequential because we never actually need the tags to get the function out. 

I propose we drop all those names. The constructor name documents the intent as it is for 100%.

Also we can't use a single name like "callback" because different types must have different names.

Reply:
In general we need a Rascal coding standard for naming callback functions/closures.

Reply:
ok I dropped the names